File: medium/1910.Remove All Occurences of a Substring.cpp
#include <string>

class Solution {
public:
    std::string removeOccurrences(std::string s, std::string part) {
        if (part.length() > s.length()) {
            return s;
        }
        
        size_t index;
        while ((index = s.find(part)) != std::string::npos) {
            s.erase(index, part.length());
        }
        return s;
    }
};

File: medium/1910.Remove All Occurences of a Substring.py
# stack
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        stk=[]
        n=len(part)
        part=list(part)
        for c in s:
            stk.append(c)
            if len(stk)>=n and stk[-n:]==part:
                stk[-n:]=[]
        return "".join(stk)


class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        while part in s:
            idx = s.find(part)
            if idx != -1:
                s = s[:idx] + s[idx+len(part):]
        return s
